### PR TITLE
Configuration

### PR DESCRIPTION
**Pull request by oravi**

### CODE REVIEW

File: macros/utils/datatypes.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{%- macro type_bool() -%}
+    {{ return(adapter.dispatch('type_bool', 'elementary_data_reliability')()) }}
```
Comment (Maayan-s) [inline macros/utils/datatypes.sql:2]: Change the name of the package so our namespace will be shorter

File: macros/system/configuration/upload_edr_configuration.sql [hunk 2]
```diff
@@ -0,0 +1,133 @@
+{% macro upload_tables_configuration() %}
+    {% if execute %}
+        {% set nodes = elementary_data_reliability.get_nodes_from_graph() %}
+        {% set table_monitors = [] %}
+        {% for node in nodes | selectattr('resource_type', 'in', 'seed,source,model') -%}
```
Comment (Maayan-s) [inline macros/system/configuration/upload_edr_configuration.sql:5]: What happens if these keys are not in the node?

File: macros/system/configuration/upload_edr_configuration.sql [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            {% set table_config_dict = elementary_data_reliability.get_table_config(node) %}
+            {% if table_config_dict is not none %}
+                {% do table_monitors.append(table_config_dict) %}
+            {% endif %}
+        {% endfor %}
+        {% if table_monitors|length > 0 %}
```
Comment (Maayan-s) [inline macros/system/configuration/upload_edr_configuration.sql:11]: You can just do 
{% if table_monitors|length %}
Do you prefer to be explicit?

File: macros/system/configuration/upload_edr_configuration.sql [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            {% do elementary_data_reliability.insert_dicts_to_table(this, table_monitors) %}
+        {% endif %}
+    {% endif %}
+    {{ return('') }}
```
Comment (Maayan-s) [inline macros/system/configuration/upload_edr_configuration.sql:15]: What does the empty return do?

File: macros/system/configuration/upload_edr_configuration.sql [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        {% if edr_config is mapping %}
+            {% do res.update(edr_config) %}
+        {% endif %}
+    {% endif %}
+    {% set edr_config = node.meta.get('edr') %}
+    {% if edr_config is mapping %}
+        {% do res.update(edr_config) %}
+    {% endif %}
+    {% set edr_config_in_columns = elementary_data_reliability.get_edr_config_in_columns(node) %}
+    {% if edr_config_in_columns | length > 0 %}
```
Comment (Maayan-s) [inline macros/system/configuration/upload_edr_configuration.sql:93]: same as just |length

File: macros/system/configuration/upload_edr_configuration.sql [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        {% do res.update({'columns': edr_config_in_columns}) %}
+    {% endif %}
+    {{ return(res) }}
+{% endmacro %}
+
+{% macro get_table_config(node) %}
+    {% set table_name = elementary_data_reliability.get_table_name(node) %}
+    {% set edr_config = elementary_data_reliability.get_edr_config(node) %}
+    {% if edr_config is mapping %}
+        {% set table_monitored = edr_config.get('monitored') %}
```
Comment (Maayan-s) [inline macros/system/configuration/upload_edr_configuration.sql:103]: What happens if monitored or columns_monitored are missing?

File: macros/utils/dicts.sql [hunk 7]
```diff
@@ -0,0 +1,35 @@
+{% macro insert_dicts_to_table(table_name, dict_list) -%}
+    {% set insert_dicts_query %}
```
Comment (Maayan-s) [inline macros/utils/dicts.sql:2]: Maybe check if table exists?

File: models/configuration_and_artifacts/column_monitors_config.sql [hunk 8]
```diff
@@ -0,0 +1,10 @@
+{{
+  config(
+    materialized = 'table',
+    transient=false,
+    unique_key = 'full_name',
+    post_hook = "{% if execute %}{{ elementary_data_reliability.upload_columns_configuration() }}{% endif %}"
+  )
+}}
+
+{{ empty_table([('full_name', 'string'), ('database_name', 'string'), ('schema_name', 'string'), ('table_name', 'string'), ('column_name', 'string'), ('column_monitors', 'string')]) }}
```
Comment (Maayan-s) [inline models/configuration_and_artifacts/column_monitors_config.sql:10]: In the rest of the project I use:
'full_table_name'
'full_column_name'
'full_schema_name'

The config is not 100% part of the project, but maybe we should be consistent anyway.

File: macros/system/configuration/upload_edr_configuration.sql (hunk 2 above)
Comment (oravi) [reply macros/system/configuration/upload_edr_configuration.sql:5]: It's not possible as it means there are no models / sources / seeds and even if it's possible I believe that it should be ok as the list should be empty and then I'm checking that tables_monitors is not empty so it just won't do anything

File: macros/system/configuration/upload_edr_configuration.sql (hunk 3 above)
Comment (oravi) [reply macros/system/configuration/upload_edr_configuration.sql:11]: Yeah, I prefer to be explicit

**Pull request merged**